### PR TITLE
Refactored word_analogies_torch.ipynb

### LINKED ISSUE #355: refactor D2L pytorch notebook code
Many of the colab notebooks in the supplementary material for the deep learning chapters (13-15) use pytorch code taken from [dive into deep learning](https://d2l.ai). This code is imported and used as follows:
`!wget https://raw.githubusercontent.com/d2l-ai/d2l-en/master/d2l/torch.py -q -O d2l.py`
`import d2l`
`net = d2l.foo(...)`
All this code needs refactoring.

- Copy and paste the relevant functions from [torch.py](https://github.com/d2l-ai/d2l-en/blob/master/d2l/torch.py) into the colab notebook, so the colab is self contained, and runs without importing d2l. (Note that in many cases,  the prefix d2l should be changed to torch or omitted. eg in [accuracy()](https://github.com/d2l-ai/d2l-en/blob/master/d2l/torch.py#L221), the code uses `d2l.argmax`, but it should be changed to   `torch.argmax`. Similarly, [plot_images()](https://github.com/d2l-ai/d2l-en/blob/master/d2l/torch.py#L174) has `d2l.plt.subplots`, this should be `plt.subplots`. This is because of [this line](https://github.com/d2l-ai/d2l-en/blob/master/d2l/torch.py#L24) in torch.py, which defines the symbol `d2l` to match the name of the file, namely `torch`.) 

- Some of these functions have names that only make sense in the context of the D2L book (eg `train_ch3` for the training function defined in chapter 3).  Rename the functions to be just `train`, removing hard-coded chapter numbers.

- Simplify the training code where appropriate, so it is customized for the model being trained in each colab, rather than having a bunch of if statements to handle cases that occur in later chapters. eg [train_batch_ch13](https://github.com/d2l-ai/d2l-en/blob/master/d2l/torch.py#L1520) has a line that refers to BERT; this code is not needed in the CNN colabs and might confuse the reader.
 
- Remove any #@save tags

- be sure to call torch.manual_seed(1) after np.random.seed(1) at the top of the notebook, to ensure reproducibility

### LINKED ISSUE #373: Refactored word_analogies_torch.ipynb
Refactor word_analogies_torch.ipynb by removing d2l library.

### PR DESCRIPTION
**Pull request by animesh-007**
@murphyk I have refactored word_analogies_torch.ipynb. Could you review this PR and let me know if the changes look good to u. Then I will try to implement the same approach for other notebooks too.

Fixes #373

### CODE REVIEW

**Comment by murphyk** [comment]
@mjsML I'm not sure how to do a code review for a jupyter notebook...

**Comment by animesh-007** [comment]
I also wanted to ask should I push each notebook for review in this PR as it's gets refactored or all in once?

**Comment by murphyk** [comment]
@animesh-007 No, just push when it's finished, and you have checked it runs in colab.  Continue to follow the [contribution guidelines](https://github.com/probml/pyprobml/blob/master/CONTRIBUTING.md) for style, etc.
